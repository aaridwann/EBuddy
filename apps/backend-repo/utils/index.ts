import { DataOrError } from './utils.types';

/**
 * Generate Response
 * @param success 
 * @param dataOrError 
 * @returns 
 */
export const generateResponse = (success: boolean, dataOrError: DataOrError) => {
  if (success) {
    return {
      success: true,
      data: dataOrError,
    };
  }

  return {
    success: false,
    error: {
      code: dataOrError.code || 500,
      message: dataOrError.message || 'Unknown error',
    },
  };
};
