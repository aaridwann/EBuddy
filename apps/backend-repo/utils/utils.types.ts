export type Error = { code: number, message: string };
export type Data = { [key: string]: string | boolean | number };

export type DataOrError = Data | Error;