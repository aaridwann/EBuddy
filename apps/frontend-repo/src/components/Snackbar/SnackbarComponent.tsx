import React from 'react';
import Snackbar from '@mui/material/Snackbar';

import { Props } from './SnackbarComponent.types';

const SnackbarComponent = (props: Props) => {
  return (
    <Snackbar
      autoHideDuration={4000}
      anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
      open={props.isOpen}
      onClose={props.onClose}
      message={props.message}
      key={`${props.message}_snackbar`}
    />
  );
};

export default SnackbarComponent;