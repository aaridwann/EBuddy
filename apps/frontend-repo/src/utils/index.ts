import { AxiosError } from 'axios';
import { get } from 'lodash';

const getErrorCode = (err: Partial<AxiosError>) => {
  const errorCode = get(err, 'payload.response.data.error.code', 'INTERNAL_ERROR');
  
  return errorCode;
};

export { getErrorCode };