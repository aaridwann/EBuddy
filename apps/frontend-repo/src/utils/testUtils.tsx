import * as React from 'react';
import { render } from '@testing-library/react';

interface Configs {
  props: {[key: string]: unknown},
  desc: string
}

export const assertSnapshot = (configs: Configs[], Component: React.ElementType) => {
  configs.map(({ props, desc }) => (
    it(desc, () => {
      expect(render(<Component {...props} />)).toMatchSnapshot();
    })
  ));
};