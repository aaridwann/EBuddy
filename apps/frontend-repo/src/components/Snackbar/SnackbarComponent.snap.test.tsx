import { assertSnapshot } from '@/utils/testUtils';

import SnackbarComponent from './SnackbarComponent';

describe('SnackbarComponent snap testing', () => {
  const configs = [
    {
      props: {
        onClose: jest.fn(),
        isOpen: true,
        message: '',  
      },
      desc: 'should render when no message',
    },
    {
      props: {
        onClose: jest.fn(),
        isOpen: true,
        message: 'something error',  
      },
      desc: 'should render when no message',
    },
  ];

  assertSnapshot(configs, SnackbarComponent);
});