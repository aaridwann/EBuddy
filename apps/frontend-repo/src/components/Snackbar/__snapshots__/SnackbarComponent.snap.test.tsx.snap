// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarComponent snap testing should render when no message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-eg3vbm-MuiSnackbar-root"
        role="presentation"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1wckuhe-MuiPaper-root-MuiSnackbarContent-root"
          direction="down"
          role="alert"
          style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message css-ozrxnr-MuiSnackbarContent-message"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-eg3vbm-MuiSnackbar-root"
      role="presentation"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1wckuhe-MuiPaper-root-MuiSnackbarContent-root"
        direction="down"
        role="alert"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message css-ozrxnr-MuiSnackbarContent-message"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SnackbarComponent snap testing should render when no message 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-eg3vbm-MuiSnackbar-root"
        role="presentation"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1wckuhe-MuiPaper-root-MuiSnackbarContent-root"
          direction="down"
          role="alert"
          style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message css-ozrxnr-MuiSnackbarContent-message"
          >
            something error
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-eg3vbm-MuiSnackbar-root"
      role="presentation"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1wckuhe-MuiPaper-root-MuiSnackbarContent-root"
        direction="down"
        role="alert"
        style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message css-ozrxnr-MuiSnackbarContent-message"
        >
          something error
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
